### ISSUE TITLE
Anyone reproduced the results by training the model using this code?

### ISSUE THREAD

**Issue by SHMCU** [nv-tlabs/GSCNN#46]
Did anyone reproduce the results by training the model using this code?
Looks like there are a few bugs and discrepancies between the code and the paper. At least, in the paper, there is a argmax ops in the loss (eq. (4) and (5)), but the code does not have argmax ops. Based on the code, eq. (5) is conducted on the output of gumbel softmax. Is this a bug or typo?

Thank you,
Hai

**Comment by tovacinni** [comment]
Thanks for your interest in our work!

We're aware of the bugs, and we're working on a patch to release soon.

In terms of the content on the paper, this should have been explained better but in the forward pass too we approximate argmax with gumbel softmax.

**Comment by SHMCU** [comment]
Thanks for sharing this code and the excellent work!
I am still having questions:
How to approximate argmax with gumbel softmax? The output of the argmax should be one channel with pixels taking values of the labels. However, the gumbel softmax in the released code has 19 channels. There seems a soft argmax ops missing. My question is that is eq.(5) computed over boundary map (output of eq. (4)) with shape (H, W,,1) or with shape (H, W, 19)? In the paper, it says (H, W, 1), but in the code, it is (H, W, 19).

Thanks!

**Comment by dreamPoet** [comment]
Have u done the training process successfully now? In fact I meet problem when I try evaluating it...the codes stuck when import encoding and did not return any error message...

**Comment by SHMCU** [comment]
> Have u done the training process successfully now? In fact I meet problem when I try evaluating it...the codes stuck when import encoding and did not return any error message...

I can train it but have not reproduced the results yet. It took four GPUs to train and I do not have so many GPUs yet.

**Comment by dreamPoet** [comment]
> > Have u done the training process successfully now? In fact I meet problem when I try evaluating it...the codes stuck when import encoding and did not return any error message...
> 
> I can train it but have not reproduced the results yet. It took four GPUs to train and I do not have so many GPUs yet.

Would you mind giving me the version of libs you used? I met problem when running import encoding... thx sooooo much!

**Comment by lxtGH** [comment]
No I didn't reproduce the results using this code.

**Comment by SHMCU** [comment]
I have moved forward. For this paper, I learned their interesting ideas. Thanks for their contribution. 
Best    On Thursday, December 26, 2019, 12:02:47 AM PST, dreamPoet <notifications@github.com> wrote:  
 
 


Have u done the training process successfully now? In fact I meet problem when I try evaluating it...the codes stuck when import encoding and did not return any error message...


I can train it but have not reproduced the results yet. It took four GPUs to train and I do not have so many GPUs yet.


Would you mind giving me the version of libs you used? I met problem when running import encoding... thx sooooo much!
